src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -642,6 +642,15 @@ ReactDOMComponent.Mixin = {
           this._currentElement.type
         );
       }
+      // Check if the component is using shady-dom
+      // see discussion in https://github.com/facebook/react/issues/7325
+      if (__DEV__) {
+        warning(
+        !el.shadyRoot,
```

Comment:
Is there a faster way to do this check? I'm hesitant to add code that touches every DOM node we create for a warning that fires a fraction of a percentage of the time.

cc @jimfb